kafkakewl-domain/src/main/scala/com/mwam/kafkakewl/domain/config/HttpConfig.scala

```diff
@@ -7,5 +7,8 @@
 package com.mwam.kafkakewl.domain.config
 
 final case class HttpConfig(
-    port: Int = 8080
+    port: Int = 8080,
+    timeout: Timeout = Timeout(10)
```

Comment:
Why not just use a zio Duration here? HOCON parsing should work I guess and it would simplify it slightly, one less custom type needed.

I guess the reason of a separate type was to be able to make it implicit/given, but perhaps the whole thing can be simplified by:
- no need for Timeout and TimeoutException classes
- no implicits of these in the endpoints
- passing the failure and the timeout with explicit parameters to the EndpointUtils.timeout() - actually we can even store a partially applied version of EndpointUtils.timeout() in the endpoints where the timeout and the error is already populated, so that we don't need to do it again at every call.

Reply:
Very much agreed, Upon coming back to it, I found a better solution. Please see the changes